### PR TITLE
Add onOpenUri

### PR DESCRIPTION
**Pull request by S-H-Y-A**
Added "onOpenUri" to hook opening uri.

### CODE REVIEW

File: html-text/src/main/java/de/charlex/compose/HtmlText.kt [hunk 1]
```diff
@@ -147,7 +148,8 @@ fun HtmlText(
     maxLines: Int = Int.MAX_VALUE,
     inlineContent: Map<String, InlineTextContent> = mapOf(),
     onTextLayout: (TextLayoutResult) -> Unit = {},
-    style: TextStyle = LocalTextStyle.current
+    style: TextStyle = LocalTextStyle.current,
+    onOpenUri: (label: String, uri: String, uriHandler: UriHandler) -> Unit = { _, uri, uriHandler -> uriHandler.openUri(uri)},
```
Comment (ch4rl3x) [inline html-text/src/main/java/de/charlex/compose/HtmlText.kt:152]: Could you rename the callback to onLinkClicked?

Comment (ch4rl3x) [inline html-text/src/main/java/de/charlex/compose/HtmlText.kt:152]: Could you rename the callback to onLinkClicked?

Comment (ch4rl3x) [inline html-text/src/main/java/de/charlex/compose/HtmlText.kt:152]: Maybe better du remove UriHandler from callback and make the callback nullable with default as null. Within th function use `onLinkClicked?.invoke(annotatedString.substring(sa.start, sa.end), sa.item) ?: uriHandler.openUri(sa.item)`

File: html-text/src/main/java/de/charlex/compose/HtmlText.kt [hunk 2]
```diff
@@ -211,7 +214,8 @@ fun HtmlText(
...
```
Comment (ch4rl3x) [inline html-text/src/main/java/de/charlex/compose/HtmlText.kt:218]: Same here

File: html-text/src/main/java/de/charlex/compose/HtmlText.kt [hunk 3]
```diff
@@ -232,22 +236,36 @@ fun HtmlText(
                         .firstOrNull()
                         ?.let { sa ->
                             if (sa.tag == "url") { // NON-NLS
-                                uriHandler.openUri(sa.item)
+                                onOpenUri.invoke(
```
Comment (ch4rl3x) [inline html-text/src/main/java/de/charlex/compose/HtmlText.kt:239]: As described something like `onLinkClicked?.invoke(annotatedString.substring(sa.start, sa.end), sa.item) ?: uriHandler.openUri(sa.item)`

File: html-text/src/main/java/de/charlex/compose/HtmlText.kt [hunk 4]
```diff
@@ -232,22 +236,36 @@ fun HtmlText(
...
                 }
             })
         }.semantics {
             if (urls.size == 1) {
                 role = Role.Button
-                onClick("Link (${annotatedString.substring(urls[0].start, urls[0].end)}") {
-                    uriHandler.openUri(urls[0].item)
+                val label = annotatedString.substring(urls[0].start, urls[0].end)
+                onClick("Link ($label)") {
+                    onOpenUri.invoke(
```
Comment (ch4rl3x) [inline html-text/src/main/java/de/charlex/compose/HtmlText.kt:253]: Same here

File: html-text/src/main/java/de/charlex/compose/HtmlText.kt [hunk 5]
```diff
@@ -232,22 +236,36 @@ fun HtmlText(
...
+                    )
                     true
                 }
             } else {
                 customActions = urls.map {
-                    CustomAccessibilityAction("Link (${annotatedString.substring(it.start, it.end)})") {
-                        uriHandler.openUri(it.item)
+                    val label = annotatedString.substring(it.start, it.end)
+                    CustomAccessibilityAction("Link ($label)") {
+                        onOpenUri.invoke(
```
Comment (ch4rl3x) [inline html-text/src/main/java/de/charlex/compose/HtmlText.kt:264]: Same here

**Review by ch4rl3x — CHANGES_REQUESTED** [review]
A few changes

**Comment by S-H-Y-A** [comment]
Check please.

**Review by ch4rl3x — APPROVED** [review]

**Comment by ch4rl3x** [comment]
Solved with #12

**Pull request closed without merging**